### PR TITLE
Add example of custom text generation step in quickstart

### LINKED ISSUE #980: [DOCS] Showcase the new `TextGeneration` in a more accessible way of the docs
## Which page or section is this issue related to?

The new `TextGeneration` task can help users starting, we should add it in an easier to access place in the docs.

### PR DESCRIPTION
**Pull request by plaguss**
## Description

Add the new `TextGeneration` fields in the Quickstart example.

Closes #980

### CODE REVIEW

File: docs/sections/getting_started/quickstart.md [hunk 1]
```diff
@@ -83,12 +85,14 @@ if __name__ == "__main__":
 
 3. We define a [`LoadDataFromHub`][distilabel.steps.LoadDataFromHub] step named `load_dataset` that will load a dataset from the Hugging Face Hub, as provided via runtime parameters in the `pipeline.run` method below, but it can also be defined within the class instance via the arg `repo_id=...`. This step will produce output batches with the rows from the dataset, and the column `prompt` will be mapped to the `instruction` field.
 
-4. We define a [`TextGeneration`][distilabel.steps.tasks.TextGeneration] task named `text_generation` that will generate text based on the `instruction` field from the dataset. This task will use the [`InferenceEndpointsLLM`][distilabel.llms.InferenceEndpointsLLM] class with the model `Meta-Llama-3.1-8B-Instruct`.
+4. We define a [`TextGeneration`](https://distilabel.argilla.io/dev/components-gallery/tasks/textgeneration/) task named `text_generation` that will generate text based on the `instruction` field from the dataset. This task will use the [`InferenceEndpointsLLM`][distilabel.llms.InferenceEndpointsLLM] class with the model `Meta-Llama-3.1-8B-Instruct`.
```
Comment (gabrielmbmb) [inline docs/sections/getting_started/quickstart.md:88]: we should use mkdocs local URLs, right?

File: docs/sections/getting_started/quickstart.md [hunk 2]
```diff
@@ -83,12 +85,14 @@ if __name__ == "__main__":
...
+5. We define the [`InferenceEndpointsLLM`][distilabel.llms.InferenceEndpointsLLM] class with the model `Meta-Llama-3.1-8B-Instruct` that will be used by the [`TextGeneration`](https://distilabel.argilla.io/dev/components-gallery/tasks/textgeneration/) task. In this case, since the [`InferenceEndpointsLLM`][distilabel.llms.InferenceEndpointsLLM] is used, we assume that the `HF_TOKEN` environment variable is set.
 
-6. We connect the `load_dataset` step to the `text_generation` task using the `rshift` operator, meaning that the output from the `load_dataset` step will be used as input for the `text_generation` task.
+6. Both `system_prompt` and `template` are optional fields. The `template` must be informed as a string following the [Jinja2](https://jinja.palletsprojects.com/en/3.1.x/templates/#synopsis) template format, and the fields that appear there ("instruction" in this case, which corresponds to the default) must be informed in the `columns` attribute. The component gallery for [`TextGeneration`](https://distilabel.argilla.io/dev/components-gallery/tasks/textgeneration/) has examples to get you started.
```
Comment (gabrielmbmb) [inline docs/sections/getting_started/quickstart.md:92]: ditto

File: docs/sections/getting_started/quickstart.md [hunk 3]
```diff
@@ -30,7 +30,7 @@ pip install distilabel[hf-inference-endpoints] --upgrade
...
+In this guide we will walk you through the process of creating a simple pipeline that uses the [`InferenceEndpointsLLM`][distilabel.llms.InferenceEndpointsLLM] class to generate text. The [`Pipeline`][distilabel.pipeline.Pipeline] will load a dataset that contains a column named `prompt` from the Hugging Face Hub via the step [`LoadDataFromHub`][distilabel.steps.LoadDataFromHub] and then use the [`InferenceEndpointsLLM`][distilabel.llms.InferenceEndpointsLLM] class to generate text based on the dataset using the [`TextGeneration`](https://distilabel.argilla.io/dev/components-gallery/tasks/textgeneration/) task.
```
Comment (gabrielmbmb) [inline docs/sections/getting_started/quickstart.md:33]: ditto

**Review by gabrielmbmb — CHANGES_REQUESTED** [review]
LGTM, but we should update the URLs

File: docs/sections/getting_started/quickstart.md (hunk 1 above)
Comment (plaguss) [reply docs/sections/getting_started/quickstart.md:88]: The first points to the API Reference, while the second points to the autogenerated components gallery. I think we cannot point to the components gallery using the local URL. I think we have clearer docs for in the components gallery, but no strong opinion on which to use

Comment (plaguss) [reply docs/sections/getting_started/quickstart.md:88]: Will keep the link while we find a better way of redirecting to the components gallery 👍

**Pull request merged**